datacube/index/_datasets.py

```diff
@@ -703,3 +703,216 @@ def search_eager(self, **query):
         :rtype: list[Dataset]
         """
         return list(self.search(**query))
+
+    def get_product_time_bounds(self, product: str):
```

Comment:
this function is not covered by tests at all, please add some

Reply:
Done